src/coreclr/vm/mono/mono_coreclr.cpp

```diff
@@ -1688,15 +1706,26 @@ extern "C" EXPORT_API void EXPORT_CC mono_unity_initialize_host_apis(initialize_
 
 extern "C" EXPORT_API MonoDomain* EXPORT_CC mono_jit_init_version(const char *file, const char* runtime_version)
 {
+    if (runtime_version != NULL)
+    {
+        if (strstr(runtime_version, "return-handles-from-api"))
+            s_ReturnHandlesFromAPI = true;
```

Comment:
I think enabling by default and then having a switch to turn off would be better.  A few reasons
(1) Correct me if I'm wrong, but I think with this opt-in change the tests are not running with the real GC path.  Keeping the embedding api tests green with the real gc seems like a minimum bar before even bothering with the real gc in unity.
(2) Opt-in means that we'll have to keep `use-real-gc` in Unity until we come back here and remove this code.  If we turn it on and go with opt-out, then we can add `disable-real-gc` for this interim period while we get it turned on and once it's ready remove `disable-real-gc`